addons/web/static/src/search/utils/dates.js

```diff
@@ -156,7 +156,10 @@ export function constructDateDomain(
         selectedOptions = getSelectedOptions(referenceMoment, searchItem, selectedOptionIds);
     }
     if ("domain" in selectedOptions) {
-        return selectedOptions.domain[0];
+        return {
+            description: selectedOptions.domain[0].description,
+            domain: Domain.and([selectedOptions.domain[0].domain, searchItem.domain]),
```

Comment:
I surely miss something but I don't understand why we have to take the first element. 🤯

Reply:
This is maybe too misleading, but I chose to name the type of the custom options for which we define a domain "domain". So `selectedOptions.domain` is the array of selected options of that type, and they all look like this: `{'description': <description>, 'domain': <domain>}`

Reply:
Ahhhh okayyyy sorry. 😅